fairroot/base/steer/FairRunAna.cxx

```diff
@@ -558,7 +558,6 @@ void FairRunAna::TerminateRun()
     //   cout << ">>>------------------------------------------------<<<" << endl;
     fRootManager->LastFill();
     fRootManager->Write();
-    fRootManager->CloseSink();
```

Comment:
Sadly enough this is also part of our public API (at least in my view). So we should keep calling the `Close` member functions in the deprecation phase. So that people can actually continue using their stuff that depends on their Sink's/Source's Close member function being called.

Maybe we should replace this with something like this at least?
```cpp
    iff(fSink) {
        fSink->Close();
    }
```

Reply:
But if we do this, we will always get deprecation warning no matter what. I think deprecation means asking people to fix something to get rid of the warning. But in this case, even they fix it and put the resource closing in the dtor, they still get this deprecation warning because the calling is guaranteed in the library code.

Reply:
The deprecation warning only happens in the FairRoot "library" code while compiling it. So normal users wont be hit by it.

And we should just stop that warning with a set of
```cpp
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
    iff(fSink) {
        fSink->Close();
    }
#pragma GCC diagnostic pop
```

Once we finally remove the Close member function, we will get a nice compile error on this, and we can remove all of this.